backbone.js

```diff
@@ -693,7 +718,12 @@
         urlError();
       if (this.isNew()) return base;
       var id = this.get(this.idAttribute);
-      return base.replace(/[^\/]$/, '$&/') + encodeURIComponent(id);
+      var consume = function(base) {
+        return base.replace(/[^\/]$/, '$&/') + encodeURIComponent(id);
+      };
+      if (base instanceof Promise)
+        return base.then(consume);
```

Comment:
Might want to make a helper for this. Also `instanceof` is not the correct way to check for `Promise`s as they should be implementation independent

See https://github.com/then/is-promise